### PR TITLE
task(config): new configuration option AppHangThresholdMillis

### PR DESCRIPTION
**Pull request by rich-bugsnag**
## Goal

New configuration option AppHangThresholdMillis to set the native Cocoa option: https://docs.bugsnag.com/platforms/ios/configuration-options/#apphangthresholdmillis

Cocoa-only.

## Testing

Manually tested, no ci tests planned due to difficulty automating the apphang

### CODE REVIEW

File: src/BugsnagUnity/Configuration.cs [hunk 1]
```diff
@@ -138,6 +138,8 @@ public virtual bool AutoTrackSessions
 
         public ErrorTypes[] EnabledErrorTypes { get; set; }
 
+        public int AppHangThresholdMillis { get; set; } = -1;
```
Comment (kattrali) [inline src/BugsnagUnity/Configuration.cs:141]: This should probably be `ulong`, to match the precision of [`NSUInteger`](https://developer.apple.com/documentation/objectivec/nsuinteger).

**Review by kattrali — COMMENTED** [review]
The logic looks good :+1: - added a minor concern over integer types

File: src/BugsnagUnity/Configuration.cs (hunk 1 above)
Comment (rich-bugsnag) [reply src/BugsnagUnity/Configuration.cs:141]: ulong can't be -1, how about we split the difference and go for a long?

Comment (rich-bugsnag) [reply src/BugsnagUnity/Configuration.cs:141]: perhaps i sould also add a limit like MaximumBreadcrumbs

Comment (kattrali) [reply src/BugsnagUnity/Configuration.cs:141]: The [threshold on iOS must be > 250](https://github.com/bugsnag/bugsnag-cocoa/blob/b83847c74669f64a9afc0e7e7ade595b620c09e8/Bugsnag/include/Bugsnag/BugsnagConfiguration.h#L210-L220), so maybe zero could be the sentinel value?

Comment (rich-bugsnag) [reply src/BugsnagUnity/Configuration.cs]: Done 👍

**Review by kattrali — APPROVED** [review]

**Pull request merged**